monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPool.java

```diff
@@ -162,6 +175,7 @@ private static SingleIpHttpClient useOldClientOrCreateNew(
         // Try to find the client with the same inetAddress in the old list and reuse it or build a new one
         return oldListOfClients.stream()
                 .filter(oldClient -> oldClient.getInetAddress().getHostAddress().equals(inetAddress.getHostAddress()))
+                .filter(oldClient -> ensureFailedResponseCountOrDecommission(oldClient, serverConfiguration))
```

Comment:
`SingleIpHttpClient::isWithinFailedResponseCountThreshold`

Reply:
This way they are not logged and closed. Just filtered.